cli/manage.go

```diff
@@ -97,8 +98,19 @@ func loadTLSConfig(ca, cert, key string, verify bool) (*tls.Config, error) {
 	return config, nil
 }
 
+func setupSwarmKitDiscovery(uri string, heartbeat time.Duration, ttl time.Duration, m map[string]string) error {
+	discovery := &swarmkit.Discovery{}
+	return discovery.Initialize(uri, heartbeat, ttl, m)
+}
+
 // Initialize the discovery service.
-func createDiscovery(uri string, c *cli.Context) discovery.Backend {
+func createDiscovery(uri string, c *cli.Context, discoveryEnginePort int) discovery.Backend {
+	// this initializes and sets up a SwarmKit discovery backend in case it
+	// is used. The discovery package handles updating the parameters
+	// when discovery.New() is called, but it requires "swarmkit" to be
+	// present as a discovery backend.
+	setupSwarmKitDiscovery("", 0, 0, nil)
```

Comment:
Why do we call this here? couldn't we wrap the `discovery.New` method in a way that detects that the backend is swarmkit and performs the initialization at that time?

Reply:
I suppose we can, I just followed the pattern used by hub based discovery. Maybe that can do with a refactor as well.